eole/utils/scoring_utils.py

```diff
@@ -42,58 +42,94 @@ def translate(self, model, gpu_rank, step):
             preds (list): Detokenized predictions
             texts_ref (list): Detokenized target sentences
         """
-        # ########## #
-        # Translator #
-        # ########## #
+        # ######### #
+        # Predictor #
+        # ######### #
 
-        # Build translator from options
+        # Build predictor from options
         model_config = self.config.model
         model_config._validate_model_config()
 
         # This is somewhat broken and we shall remove or improve
         # (take 'inference' field of config if exists?)
         # Set "default" translation options on empty cfgfile
         self.config.training.num_workers = 0
+        is_seq2seq = model.encoder is not None and model.decoder is not None
+        if not is_seq2seq:
+            if "insert_mask_before_placeholder" in self.config.transforms:
+                self.response_patterns = self.config.transforms_configs.insert_mask_before_placeholder.response_patterns
+        else:
+            self.response_patterns = None
+
         predict_config = PredictConfig(
             model_path=["dummy"],
-            src=self.config.data["valid"].path_src,
+            src="dummy",
             compute_dtype=self.config.training.compute_dtype,
             beam_size=1,
             transforms=self.config.transforms,
             transforms_configs=self.config.transforms_configs,
             model=model_config,
             tgt_file_prefix=self.config.transforms_configs.prefix.tgt_prefix != "",
             gpu_ranks=[gpu_rank],
+            batch_type=self.config.training.batch_type,
+            batch_size=self.config.training.batch_size,
         )
 
         scorer = GNMTGlobalScorer.from_config(predict_config)
-        translator = Translator.from_config(  # we need to review opt/config stuff in translator
-            model,
-            self.vocabs,
-            predict_config,
-            model_config,
-            device_id=gpu_rank,
-            global_scorer=scorer,
-            report_align=predict_config.report_align,
-            report_score=False,
-            logger=None,
-        )
+
+        if is_seq2seq:
+            predictor = Translator.from_config(  # we need to review opt/config stuff in translator
+                model,
+                self.vocabs,
+                predict_config,
+                model_config,
+                device_id=gpu_rank,
+                global_scorer=scorer,
+                report_align=predict_config.report_align,
+                report_score=False,
+                logger=None,
+            )
+        else:
+            predictor = GeneratorLM.from_config(
+                model,
+                self.vocabs,
+                predict_config,
+                model_config,
+                device_id=gpu_rank,
+                global_scorer=scorer,
+                report_align=predict_config.report_align,
+                report_score=False,
+                logger=None,
+            )
```

Comment:
Maybe cleaner to just define a `predictor_class` in the condition, and call `predictor_class(*)` once, since they should have the same signature.